ts/features/messages/components/MessageDetail/MessagePaymentItem.tsx

```diff
@@ -190,17 +190,16 @@ export const MessagePaymentItem = ({
   const store = useIOStore();
   const toast = useIOToast();
 
-  const shouldUpdatePayment = shouldUpdatePaymentSelector(
-    store.getState(),
-    messageId,
-    rptId
+  const shouldUpdatePayment = useIOSelector(state =>
```

Comment:
`shouldUpdatePayment` should not be computed using a selector. Such data is used to dispatch the update action and the component should not re-render due to changes in them.

We could also consider to move such data computing directly inside the related useEffect, as to remove the external dependency from `shouldUpdatePayment`

Reply:
Yeah, good idea